tuist/tuist
# Document SPM build tool plugin requirements

### Short description 📝

SPM build tool plugins must be declared using [Xcode's default integration](https://docs.tuist.io/guide/project/dependencies.html#xcode-s-default-integration) mechanism, even when using Tuist's [XcodeProj-based integration](https://docs.tuist.io/guide/project/dependencies.html#tuist-s-xcodeproj-based-integration) for project dependencies. Because this wasn't an obvious requirement, this PR adds some documentation to help explain it.

### How to test the changes locally 🧐

The PR includes a fixture that demonstrates a working SPM build tool plugin. The reviewer can verify that the build tool plugin works both when opening the swift package, and when generating an Xcode project.

One unanswered question is whether SPM command plugins are supported. There is a "SwiftLintFix" command included in the `lukepistrol/SwiftLintPlugin` package, and it's available when you open the package via its Package.swift manifest. When you generate an Xcode project, it does not appear to be available. I don't know enough about SPM commands in an Xcode project context to say whether this is a separate issue.

### Contributor checklist ✅


### Reviewer checklist ✅
